controllers/releaseplan/controller.go

```diff
@@ -63,8 +63,8 @@ func (c *Controller) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	adapter := newAdapter(ctx, c.client, releasePlan, loader.NewLoader(), &logger)
 
 	return controller.ReconcileHandler([]controller.Operation{
-		adapter.EnsureOwnerReferenceIsSet,
 		adapter.EnsureMatchingInformationIsSet,
+		adapter.EnsureOwnerReferenceIsSet,
```

Comment:
Just checking: This has no effect on tests? Is there a way we could add a test for the scenario we're fixing? If not, it's ok.

Reply:
I'm not aware of any way to test this besides creating pretty involved tests per adapter that does everything the adapter does and ensures it happens in the right order. It would pretty much be recreating e2e

Reply:
I see. It's fine then.